collect_app/src/main/java/org/odk/collect/android/architecture/MVVMViewModel.java

```diff
@@ -0,0 +1,28 @@
+package org.odk.collect.android.architecture;
+
+import android.arch.lifecycle.ViewModel;
+
+import javax.annotation.OverridingMethodsMustInvokeSuper;
+
+/**
+ * A {@link ViewModel} subclass that provides an 'onCreate' method for simplifying
+ * subscription timing.
+ */
+public class MVVMViewModel extends ViewModel {
+
+    private boolean wasCreated = false;
```

Comment:
Is the intention here to prevent the same `ViewModel` instance from having `.create()` called on it twice if the Activity has been rotated or recreated?

Reply:
Exactly. The architecture examples show using the ViewModel's constructor for starting tasks/subscriptions, but I don't think Constructors should have side effects. This also keeps things more in line with the Android standard of having onCreate/onDestroy pairs.